cpp/es3.1/PhoneBook.cpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+struct Contact {
+    string nome;
+    string cognome;
+    string numTelefono;
+};
+
+class PhoneBook {
+private:
+    vector<Contact> contatti;
+
+public:
+    void append(const Contact& contatto) {
+        contatti.push_back(contatto);
+    }
+
+    Contact* find(const string& nome) {
+        for (auto& contatto : contatti) {
+            if (contatto.nome == nome) {
+                return &contatto;
+            }
+        }
+        return nullptr;
+    }
+
+    vector<Contact> filter(int scelta) {
```

Comment:
Secondo te possiamo modificare la funzione `filter()` in modo tale che nuovi algoritmi di filtraggio non richiedano la modifica della funzione filter?